### ISSUE TITLE
jsx-no-undef throws on multiple lowercase sections for object access

### ISSUE THREAD

**Issue by Standard8** [yannickcr/eslint-plugin-react#469]
Seems to be a regression from #435.

This warns unexpectedly:

   <foo.bar.Component />

This doesn't warn:

    <foo.Component />

The only way I can find around this is to alias the component and use the alias in the react code.

**Comment by Standard8** [comment]
Here's a diff for test cases against current master:

```
diff --git a/tests/lib/rules/jsx-no-undef.js b/tests/lib/rules/jsx-no-undef.js
index e1e9c17..355ae41 100644
--- a/tests/lib/rules/jsx-no-undef.js
+++ b/tests/lib/rules/jsx-no-undef.js
@@ -34,6 +34,15 @@ ruleTester.run('jsx-no-undef', rule, {
     code: '/*eslint no-undef:1*/ var React, App; React.render(<App />);',
     parserOptions: parserOptions
   }, {
+    code: '/*eslint no-undef:1*/ var React, app; React.render(<app />);',
+    parserOptions: parserOptions
+  }, {
+    code: '/*eslint no-undef:1*/ var React, app; React.render(<app.Foo />);',
+    parserOptions: parserOptions
+  }, {
+    code: '/*eslint no-undef:1*/ var React, app; React.render(<app.fake.Foo />);',
+    parserOptions: parserOptions
+  }, {
     code: '/*eslint no-undef:1*/ var React; React.render(<img />);',
     parserOptions: parserOptions
   }, {
```

The `app.fake.Foo` test fails.

**Comment by ljharb** [comment]
I've seen this with `<this.state.Handler />` as well
